portafly/src/components/AppLayout.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react'
+import {
+  Nav,
+  NavList,
+  NavVariants,
+  Page,
+  PageHeader,
+  PageSidebar,
+  SkipToContent,
+  PageHeaderProps,
+} from '@patternfly/react-core'
+import {AppNavExpandable, IAppNavExpandableProps} from './AppNavExpandable'
+import {AppNavGroup, IAppNavGroupProps} from './AppNavGroup'
+import {AppNavItem, IAppNavItemProps} from './AppNavItem'
+
+export interface IAppLayoutContext {
+  setBreadcrumb: (breadcrumb: React.ReactNode) => void
+}
+
+export const AppLayoutContext = React.createContext<IAppLayoutContext>({
+  setBreadcrumb: () => void 0,
+})
+
+export interface IAppLayoutProps
+  extends Pick<PageHeaderProps, 'logo'>,
+    Pick<PageHeaderProps, 'logoProps'>,
+    Pick<PageHeaderProps, 'avatar'>,
+    Pick<PageHeaderProps, 'toolbar'> {
+  navVariant?: 'vertical' | 'horizontal'
+  navItems?: Array<IAppNavItemProps | IAppNavExpandableProps | IAppNavGroupProps | undefined>
+  navGroupsStyle?: 'grouped' | 'expandable'
+  startWithOpenNav?: boolean
+  theme?: 'dark' | 'light'
+  mainContainerId?: string
+}
+
+export const AppLayout: React.FunctionComponent<IAppLayoutProps> = ({
+  logo,
+  logoProps,
+  navVariant = 'horizontal',
+  navItems = [],
+  navGroupsStyle = 'grouped',
+  toolbar,
+  avatar,
+  startWithOpenNav = true,
+  theme = 'dark',
+  mainContainerId = 'main-container',
+  children,
+}) => {
+  const [isNavOpen, setIsNavOpen] = React.useState(startWithOpenNav)
+  const [isMobileView, setIsMobileView] = React.useState(true)
+  const [isNavOpenMobile, setIsNavOpenMobile] = React.useState(false)
+  const [breadcrumb, setBreadcrumb] = React.useState<React.ReactNode | undefined>()
+  const previousBreadcrumb = React.useRef<React.ReactNode | null>()
+
+  const handleSetBreadcrumb = React.useCallback(
+    (newBreadcrumb: React.ReactNode) => {
+      if (previousBreadcrumb.current !== newBreadcrumb) {
+        previousBreadcrumb.current = newBreadcrumb
+        setBreadcrumb(previousBreadcrumb.current)
+      }
+    },
+    [setBreadcrumb, previousBreadcrumb]
+  )
+
+  const onNavToggleMobile = React.useCallback(() => {
+    setIsNavOpenMobile(!isNavOpenMobile)
+  }, [setIsNavOpenMobile, isNavOpenMobile])
+
+  const onNavToggle = React.useCallback(() => {
+    setIsNavOpen(!isNavOpen)
+  }, [setIsNavOpen, isNavOpen])
+
+  const onPageResize = (props: { mobileView: boolean; windowSize: number }) => {
```

Comment:
Since we're not using `windowSize`, should we still define it?
```suggestion
  const onPageResize = ({ mobileView } : { mobileView: boolean }) => {
```

Reply:
It seems something got lost in definition heehehe